### PR TITLE
[Silabs] Silabs Refactor build system for 917

### PR DESCRIPTION
**Pull request by jepenven-silabs**
PR 1 for deleting the duplicated examples code

### CODE REVIEW

File: scripts/examples/gn_efr32_example.sh [hunk 1]
```diff
@@ -169,6 +171,7 @@ else
                     exit 1
                 fi
                 USE_WIFI=true
+                optArgs+="chip_device_platform =\"efr32\" "
```
Comment (jmartinez-silabs) [inline scripts/examples/gn_efr32_example.sh:174]: You do it later on if the other ifs line 255 and 272. Same for the `USE_WIFI=true`

You can delete this one

File: examples/window-app/silabs/efr32/BUILD.gn [hunk 2]
```diff
@@ -19,57 +19,85 @@ import("//build_overrides/pigweed.gni")
...
+        "${examples_plat_dir}/rs911x",
+        "${examples_plat_dir}/rs911x/hal",
+      ]
+    }
   }
 }
 
 silabs_executable("window_app") {
-  output_name = "chip-efr32-window-example.out"
+  output_name = "chip-silabs-window-example.out"
```
Comment (jmartinez-silabs) [inline examples/window-app/silabs/efr32/BUILD.gn:100]: do we want to keep `chip_` might as well change it while we are here.

**Review by jmartinez-silabs — APPROVED** [review]

**Review by bzbarsky-apple — APPROVED** [review]
Rubber-stamping.

File: examples/window-app/silabs/efr32/BUILD.gn (hunk 2 above)
Comment (jepenven-silabs) [reply examples/window-app/silabs/efr32/BUILD.gn:100]: changed

File: scripts/examples/gn_efr32_example.sh (hunk 1 above)
Comment (jepenven-silabs) [reply scripts/examples/gn_efr32_example.sh:174]: Nope we need to keep this one. 
This one is for the build option `--wifi 9116`
Line 255 if for the build argument `use_rs9116=true`
And last one is for the 917. 

i know it's a lot of duplication but they are all needed to support the same behaviour that we once had.

File: third_party/silabs/BUILD.gn [hunk 3]
```diff
@@ -26,8 +26,8 @@ if (silabs_board == "BRD4325B") {  # CCP board
 
 declare_args() {
   # Build target to use for efr32 SDK. Use this to set global SDK defines.
-  efr32_sdk_target = ""
-  siwx917_sdk_target = ""
+  silabs_sdk_target = ""
+  silabs_sdk_target = ""
```
[the remark this answers is not in the document]
Comment (jepenven-silabs) [reply third_party/silabs/BUILD.gn]: removed

**Pull request merged**